sound/soc/sof/sof-priv.h

```diff
@@ -308,6 +309,9 @@ struct snd_sof_dev {
 	int ipc_irq;
 	u32 next_comp_id; /* monotonic - reset during S3 */
 
+	/* hda bus */
```

Comment:
do we really want hda_bus at the snd_sof_dev level? This is a bit Intel-centric, maybe it should be hidden in HDA specific structures, no?

Reply:
today I will work on this, Liam suggest me to unify it with sof_intel_hda_dev(delete  sof_intel_hda_dev if possible), Mengdong give a good suggestion that using private struct for it then other arch(e.g. ARM, non-HDA stuff) can reuse it.